tests/fixtures/sorting_maybe_nesting_related/foo.rs

```diff
@@ -0,0 +1,72 @@
+impl TransformTypeScript {
```

Comment:
I tried making a little "dummy" test case that I thought had the relevant quality (of the outermost matched AST node being in the reverse order wrt the order of the AST nodes for the capture we specified) but it didn't in fact fail so maybe my understanding of in what cases we were getting "out-of-order" matches wasn't correct

But this is a "real" example that was actually failing (and reproducibly fails here if you don't explicitly sort the matches)